chatsky/llm/wrapper.py

```diff
@@ -0,0 +1,203 @@
+    """
+
+    def __init__(
+        self,
+        model: BaseChatModel,
+        system_prompt: str = "",
+    ) -> None:
+        """
+        :param model: Model object.
+        :param system_prompt: System prompt for the model.
+        """
+        self.__check_imports()
+        self.model: BaseChatModel = model
+        self.name = ""
+        self.parser = StrOutputParser()
+        self.system_prompt = system_prompt
+
+    def __check_imports(self):
+        if not langchain_available:
+            raise ImportError("Langchain is not available. Please install it with `pip install chatsky[llm]`.")
+
+    async def respond(
+        self, history: list = [""], message_schema: Union[None, Type[Message], Type[BaseModel]] = None
+    ) -> Message:
+        if message_schema is None:
+            result = await self.parser.ainvoke(await self.model.ainvoke(history))
+            result = Message(text=result)
+
+        elif issubclass(message_schema, Message):
+            # Case if the message_schema desribes Message structure
+            structured_model = await self.model.with_structured_output(message_schema)
+            result = Message.model_validate(await structured_model.invoke(history))
+        elif issubclass(message_schema, BaseModel):
+            # Case if the message_schema desribes Message.text structure
+            structured_model = await self.model.with_structured_output(message_schema)
+            result = await structured_model.invoke(history)
+            result = Message(text=str(result.json()))
+
+        if result.annotations:
+            result.annotations["__generated_by_model__"] = self.name
+        else:
+            result.annotations = {"__generated_by_model__": self.name}
+        return result
+
+    async def condition(self, prompt: str, method: BaseMethod):
+        async def process_input(ctx: Context, _: Pipeline) -> bool:
+            condition_history = [await message_to_langchain(Message(prompt), pipeline=_, source="system"),
+                await message_to_langchain(ctx.last_request, pipeline=_, source="human")]
+            result = method(ctx, await self.model.agenerate([condition_history], logprobs=True, top_logprobs=10))
+            return result
+
+        return process_input
+
+
+def llm_response(model_name: str, prompt: str = "", history: int = 5, filter_func: Callable = lambda *args: True, message_schema: Union[None, Type[Message], Type[BaseModel]] = None):
+    """
+    Basic function for receiving LLM responses.
+    :param ctx: Context object. (Assigned automatically)
+    :param pipeline: Pipeline object. (Assigned automatically)
+    :param model_name: Name of the model from the `Pipeline.models` dictionary.
+    :param prompt: Prompt for the model.
+    :param history: Number of messages to keep in history. `-1` for full history.
+    :param filter_func: filter function to filter messages that will go the models context.
+    """
```

Comment:
Add annotation that this raises `ValueError` if any message is longer than the limit.